posthog/views.py

```diff
@@ -58,15 +59,48 @@ def handler(request, *args, **kwargs):
 
 
 def health(request):
-    executor = MigrationExecutor(connections[DEFAULT_DB_ALIAS])
-    plan = executor.migration_plan(executor.loader.graph.leaf_nodes())
-    status = 503 if plan else 200
-    if status == 503:
+    """
+    Validate that everything this process need to operate correctly is in place.
+    Returns a dict of checks to boolean status, returning 503 status if any of
+    them is non-True
+    """
+    migrations_uptodate = health_migrations()
+    kafka_connected = health_kafka()
+
+    if migrations_uptodate == False:
+        # NOTE: before adding kafka to this health check, we were just checking
+        # migrations, sending the below sentry error, and returning 503. I've
+        # left this sentry error here but it can probably be removed, and issues
+        # with migration alerting managed elsewhere.
         err = Exception("Migrations are not up to date. If this continues migrations have failed")
         sentry_sdk.capture_exception(err)
-        return HttpResponse("Migrations are not up to date", status=status, content_type="text/plain")
-    if status == 200:
-        return HttpResponse("ok", status=status, content_type="text/plain")
+
+    status = 200 if migrations_uptodate and kafka_connected else 500
+    return JsonResponse({"migrations_uptodate": migrations_uptodate, "kafka_connected": kafka_connected}, status=status)
```

Comment:
Do we have any services that currently look for the `ok` in the response? I'd hope they look for the 200 only but just wondering

Reply:
I've decided to avoid updating the existing endpoint entirely, and rather use endpoints specific to the purpose of k8s readiness and liveness checks